x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/rules/logical/DropIndexMode.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.esql.optimizer.rules.logical;
+
+import org.elasticsearch.index.IndexMode;
+import org.elasticsearch.xpack.esql.core.expression.MetadataAttribute;
+import org.elasticsearch.xpack.esql.plan.logical.EsRelation;
+import org.elasticsearch.xpack.esql.plan.logical.LogicalPlan;
+
+/**
+ * Uses the standard index mode if the special index mode is not required in the query.
+ */
+public final class DropIndexMode extends OptimizerRules.OptimizerRule<EsRelation> {
```

Comment:
how about `PruneUnusedIndexMode` ?

Reply:
renamed in https://github.com/elastic/elasticsearch/pull/125193/commits/aa3a26c62d85fdd2e834968cc9eab90df5db3e86